atlas/packages/components/src/components/Carousel/Carousel.tsx

```diff
@@ -1,39 +1,105 @@
-import React, { ReactNode, useRef } from "react";
+import React, { useState, useRef, useEffect, useCallback, useMemo } from "react";
+import { css, SerializedStyles } from "@emotion/core";
+import { animated, useSpring } from "react-spring";
+import useResizeObserver from "use-resize-observer";
 import { useCSS, CarouselStyleProps } from "./Carousel.style";
 import NavButton from "../NavButton";
 
 type CarouselProps = {
-	children: Array<ReactNode>;
-	scrollAmount?: Number;
+	children: React.ReactNode;
+	containerCss: SerializedStyles;
+	leftControlCss: SerializedStyles;
+	rightControlCss: SerializedStyles;
+	onScroll: (direction: "left" | "right") => void;
 } & CarouselStyleProps;
 
-export default function Carousel({ children, scrollAmount = 200, ...styleProps }: CarouselProps) {
-	const container = useRef(null);
-
-	function onScroll(direction: "right" | "left") {
-		container.current.scrollBy({
-			left: direction === "left" ? -scrollAmount : scrollAmount,
-			behavior: "smooth",
-		});
+const Carousel: React.FC<Partial<CarouselProps>> = ({
+	children,
+	containerCss,
+	leftControlCss,
+	rightControlCss,
+	onScroll = () => {},
+}) => {
+	if (!Array.isArray(children)) {
+		return <>{children}</>;
 	}
+	let [props, set] = useSpring(() => ({
+		transform: `translateX(0px)`,
+	}));
+	const [x, setX] = useState(0);
+	const { width: containerWidth, ref: containerRef } = useResizeObserver<HTMLDivElement>();
+	const elementsRefs = useRef<(HTMLDivElement | null)[]>([]);
+	const [childrenLength, setChildrenLength] = useState(0);
```

Comment:
What is this variable? Looking at the `useEffect` it looks like total width of all the items but the name suggests it's a length (of an array my first thought was)